src/util/shared-prop-types.js

```diff
@@ -14,12 +14,33 @@
  */
 
 import PropTypes from 'prop-types';
+import deprecate from './deprecate';
 import { TOP, BOTTOM, LEFT, RIGHT, START, END, CENTER } from './constants';
 
 // TODO: figure out if we can still get these props in react-docgen
 //       when they are imported and merged into a component's
 //       propTypes.
 
+export const deprecatedPropType = (propType, explanation = '') => (
+  props,
+  propName,
+  componentName
+) => {
+  if (props[propName] !== null) {
+    deprecate(
+      // eslint-disable-next-line max-len
+      `"${propName}" prop of "${componentName}" has been deprecated.\n${explanation}`
+    );
+  }
+
+  return PropTypes.checkPropTypes(
+    { propName: propType },
```

Comment:
@connor-baer just noticed this, shouldn't it be `{ [propName]: propType }`?

Reply:
You're right! I copied it from another custom prop type where it's also wrong. I'll make a PR to fix it 👍

Reply:
Here you go: https://github.com/sumup/circuit-ui/pull/449

Reply:
approved1